### PR TITLE
EDM-2916: Extend Podman Client with Image and Artifact Management Methods

### PR DESCRIPTION
**Pull request by ori-amizur**
## Summary by CodeRabbit

* **New Features**
  * Added image listing and removal capabilities for container images (deduplicated, supports tagged and untagged formats)
  * Added artifact listing and removal capabilities for OCI artifacts (checks for artifact support before listing)

* **Reliability**
  * Non-fatal handling for attempts to remove non-existent images/artifacts
  * Per-call timeouts applied to list/remove operations

* **Tests**
  * Added comprehensive unit tests covering listing, removal, edge cases, and error handling

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/agent/client/podman.go [hunk 1]
```diff
@@ -314,6 +314,113 @@ func (p *Podman) InspectArtifactAnnotations(ctx context.Context, reference strin
...
+// Returns image references in the format "repository:tag" or image ID for untagged images.
+func (p *Podman) ListImages(ctx context.Context) ([]string, error) {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
+	defer cancel()
+
+	// Use a format that handles both tagged and untagged images
+	// For tagged images: "repository:tag"
+	// For untagged images: use the image ID (short format)
+	// Note: When Repository is the string "<none>" (not empty), we need to check for it explicitly
+	args := []string{"image", "ls", "--format", "{{if and .Repository (ne .Repository \"<none>\")}}{{.Repository}}:{{.Tag}}{{else}}{{.ID}}{{end}}"}
```
Comment (hexfusion) [inline internal/agent/client/podman.go:334]: question: when would we have a non tagged image, is this nessisary?

File: internal/agent/client/podman.go [hunk 2]
```diff
@@ -314,6 +314,113 @@ func (p *Podman) InspectArtifactAnnotations(ctx context.Context, reference strin
...
+// ListArtifacts returns a list of all OCI artifacts stored on the device.
+func (p *Podman) ListArtifacts(ctx context.Context) ([]string, error) {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
+	defer cancel()
+
+	if err := p.EnsureArtifactSupport(ctx); err != nil {
+		return nil, err
+	}
+
+	args := []string{"artifact", "ls", "--format", "{{.Name}}"}
```
Comment (hexfusion) [inline internal/agent/client/podman.go:366]: we need to be careful with artifact support what version of podman was this added? can we review that with what is shipped with RHEL 9.5+. we don't want issues where support is not available in some versions uncontrolled. this probably is true for other areas of the client but lets verify :pray:

File: internal/agent/client/podman.go [hunk 3]
```diff
@@ -314,6 +314,113 @@ func (p *Podman) InspectArtifactAnnotations(ctx context.Context, reference strin
...
+// Returns an error if the removal fails. Handles non-existent images gracefully.
+func (p *Podman) RemoveImage(ctx context.Context, image string) error {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
+	defer cancel()
+
+	args := []string{"image", "rm", image}
+	_, stderr, exitCode := p.exec.ExecuteWithContext(ctx, podmanCmd, args...)
+	if exitCode != 0 {
+		// Check if error is due to image not existing
+		if strings.Contains(stderr, "no such image") || strings.Contains(stderr, "image not known") {
```
Comment (hexfusion) [inline internal/agent/client/podman.go]: can we make these strings consts please

File: internal/agent/client/podman.go [hunk 4]
```diff
@@ -314,6 +314,113 @@ func (p *Podman) InspectArtifactAnnotations(ctx context.Context, reference strin
...
+	return nil
+}
+
+// RemoveArtifact removes the specified OCI artifact from Podman.
+// Returns an error if the removal fails. Handles non-existent artifacts gracefully.
+func (p *Podman) RemoveArtifact(ctx context.Context, artifact string) error {
+	ctx, cancel := context.WithTimeout(ctx, p.timeout)
+	defer cancel()
+
+	if err := p.EnsureArtifactSupport(ctx); err != nil {
```
Comment (hexfusion) [inline internal/agent/client/podman.go:414]: :+1: ensure this is used on all artifact calls

File: internal/agent/client/podman.go [hunk 5]
```diff
@@ -314,6 +314,113 @@ func (p *Podman) InspectArtifactAnnotations(ctx context.Context, reference strin
...
+		return err
+	}
+
+	args := []string{"artifact", "rm", artifact}
+	_, stderr, exitCode := p.exec.ExecuteWithContext(ctx, podmanCmd, args...)
+	if exitCode != 0 {
+		// Check if error is due to artifact not existing
+		if strings.Contains(stderr, "no such artifact") || strings.Contains(stderr, "artifact not known") {
```
Comment (hexfusion) [inline internal/agent/client/podman.go]: consts please

File: internal/agent/client/podman.go (hunk 1 above)
Comment (ori-amizur) [reply internal/agent/client/podman.go:334]: It is a generic function.  For pruning there is no need.

File: internal/agent/client/podman.go (hunk 4 above)
Comment (ori-amizur) [reply internal/agent/client/podman.go:414]: It is already used

File: internal/agent/client/podman.go (hunk 2 above)
Comment (ori-amizur) [reply internal/agent/client/podman.go:366]: In general, all artifact functions verify artifact capabilities

**Review by hexfusion — APPROVED** [review]
